addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2021 Red Hat OpenShift Data Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"context"
+	"fmt"
+
+	replicationv1alpha1 "github.com/csi-addons/volume-replication-operator/api/v1alpha1"
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/common"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// MirrorPeerReconciler reconciles a MirrorPeer object
+type MirrorPeerReconciler struct {
+	HubClient        client.Client
+	Scheme           *runtime.Scheme
+	SpokeClient      client.Client
+	SpokeClusterName string
+}
+
+const (
+	RookCSIEnableKey              = "CSI_ENABLE_OMAP_GENERATOR"
+	RookVolumeRepKey              = "CSI_ENABLE_VOLUME_REPLICATION"
+	MirroringModeKey              = "mirroringMode"
+	SchedulingIntervalKey         = "schedulingInterval"
+	ReplicationSecretNameKey      = "replication.storage.openshift.io/replication-secret-name"
+	ReplicationSecretNamespaceKey = "replication.storage.openshift.io/replication-secret-namespace"
+	Provisioner                   = "openshift-storage.rbd.csi.ceph.com"
```

Comment:
This is not a constant. Provisioner name depends on the namespace of the cluster.